src/openapi/streaming/connection/transport/websocket-transport.spec.ts

```diff
@@ -348,6 +349,108 @@ describe('openapi WebSocket Transport', () => {
+                transport.updateQuery(AUTH_TOKEN, CONTEXT_ID);
+                transport.start({}, spyOnStartCallback);
+                fetchMock.resolve(200, {});
+
+                transport.authorizePromise?.then(() => {
+                    const dataBuffer = new window.TextEncoder().encode(
+                        JSON.stringify(jsonPayload),
+                    );
+                    const payload = new Uint8Array(
+                        new ArrayBuffer(dataBuffer.length + 17),
+                    );
+                    payload.set(
+                        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 56, 0, 134, 12, 0, 0],
+                        0,
+                    );
+                    payload.set(dataBuffer, 17);
+                    transport.socket?.onmessage?.({
+                        data: payload.buffer,
+                    } as any);
+
+                    // send another message with same id i.e. 1
+                    transport.socket?.onmessage?.({
+                        data: payload.buffer,
+                    } as any);
+
+                    expect(logSpy).toBeCalledWith(
+                        'PlainWebSocketsTransport',
+                        'Messages out of order in websocket transport',
+                        { lastMessageId: 1, messageId: 1 },
+                    );
+                    done();
+                });
+            });
+
+            it('should skip _connectionheartbeat control message from incremental id logic', (done) => {
+                const logSpy = jest.spyOn(log, 'error');
+                const spyOnReceivedCallback = jest
+                    .fn()
+                    .mockName('spyReceivedCallback');
+                const spyOnStartCallback = jest
+                    .fn()
+                    .mockName('spyStartCallback');
+
+                const transport = new WebSocketTransport(BASE_URL);
+                transport.setReceivedCallback(spyOnReceivedCallback);
+                transport.updateQuery(AUTH_TOKEN, CONTEXT_ID);
+                transport.start({}, spyOnStartCallback);
+                fetchMock.resolve(200, {});
+
+                transport.authorizePromise?.then(() => {
+                    const dataBuffer = new window.TextEncoder().encode(
+                        JSON.stringify(jsonPayload),
+                    );
+                    const payload = new Uint8Array(
+                        new ArrayBuffer(dataBuffer.length + 17),
+                    );
+                    payload.set(
+                        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 56, 0, 134, 12, 0, 0],
+                        0,
+                    );
+                    payload.set(dataBuffer, 17);
+                    transport.socket?.onmessage?.({
+                        data: payload.buffer,
+                    } as any);
+
+                    // send _connectionheartbeat control message with same id i.e. 1

```

Comment:
where is the message id 1? is it in payload2.set below?

Reply:
yes,
the first field is the message id 
```
payload2.set(
                        [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 56, 0, 40, 0, 0, 0],
                        0,
                    );
                    ```